app/src/main/java/me/quadphase/qpdex/pokedex/CentralAudioPlayer.java

```diff
@@ -0,0 +1,113 @@
+package me.quadphase.qpdex.pokedex;
+
+import android.content.res.AssetFileDescriptor;
+import android.media.MediaPlayer;
+import android.util.Log;
+
+/**
+ * Handles all interaction related to Android's MediaPlayer Class to server Pokemon Cries correctly
+ * This class is a Singleton that gets instantiated by the {@link PokedexManager}
+ */
+//TODO: Document with JavaDocs!
+public class CentralAudioPlayer implements IMediaPlayerWrapper {
+    //Internal Class
+    protected class CompletedPlayback implements MediaPlayer.OnCompletionListener{
+        @Override
+        public void onCompletion(MediaPlayer mediaPlayer){
+            //Go back to the begginning of the track if nothing has changed.
+            isPlaying = false;
+            if(isDirty){
+                mediaPlayer.reset();
+                setNewPokemonCry();
+            }
+            else{
+                mediaPlayer.seekTo(0);
+            }
+        }
+    }
+
+    //All Private Variables
+
+    private static CentralAudioPlayer instance = null;
+
+    private MediaPlayer androidMP;
+    private AssetFileDescriptor cachedAudioFile=null;
+    private int cachedNationalID =0;
+    private boolean isDirty=true;
+    private boolean isPlaying=false;
+
+
+    //Constructor and Instance Access for Singleton
+    protected CentralAudioPlayer(){
+        androidMP = new MediaPlayer();
+        androidMP.setOnCompletionListener(new CompletedPlayback());
+    }
+
+    public static CentralAudioPlayer getInstance(){
+        if(instance==null){
+            instance = new CentralAudioPlayer();
+        }
+        return instance;
+    }
+
+    private void setNewPokemonCry(){
+        try {
+            androidMP.setDataSource(cachedAudioFile.getFileDescriptor(),
+                    cachedAudioFile.getStartOffset(),
+                    cachedAudioFile.getLength());
+        }
+        catch (java.io.IOException exception){
+            Log.e("QPDEX",exception.getMessage()+"\nCentral Media Player is Dirty and needs to Update Instance");
+            isDirty = true;
+            isPlaying = false;
+        }
+    }
+
+
+    //Methods for behaviour
+    @Override
+    public void updateInstace(int pokemonNationalID, AssetFileDescriptor pokemonCry){
```

Comment:
there is a typo in updateInstace->updateInstance.
Since you are overriding the method, I'm assuming that this is intentional because it is typo-ed in the IMediaPlayerWrapper class, or needs to be fixed.